### ISSUE TITLE
Spam of push notifications on iOS 17 Beta

### ISSUE THREAD

**Issue by kebuenokebueno** [kreait/firebase-php#814]
### Describe the bug

Hello, I am testing iOS 17 Beta with a script that sums up what we do in our app. I can add it if required. Let me say that it does work fine (single push) when running on iOS 16.

### Installed packages

```json
{
    "name": "XXX/messagetest",
    "type": "project",
    "require": {
        "kreait/firebase-php": "^7.5"
    },
    "autoload": {
        "psr-4": {
            "XXX\\Messagetest\\": "src/"
        }
    },
    "authors": [
        {
            "name": "XXX YYY",
            "email": "ZZZZZ"
        }
    ]
}
```

### PHP version and extensions

```shell
composer                           2.5.8    Composer package
composer-plugin-api                2.3.0    The Composer Plugin API
composer-runtime-api               2.2.2    The Composer Runtime API
ext-bcmath                         8.2.7    The bcmath PHP extension
ext-bz2                            8.2.7    The bz2 PHP extension
ext-calendar                       8.2.7    The calendar PHP extension
ext-ctype                          8.2.7    The ctype PHP extension
ext-curl                           8.2.7    The curl PHP extension
ext-date                           8.2.7    The date PHP extension
ext-dba                            8.2.7    The dba PHP extension
ext-dom                            20031129 The dom PHP extension
ext-exif                           8.2.7    The exif PHP extension
ext-ffi                            8.2.7    The FFI PHP extension
ext-fileinfo                       8.2.7    The fileinfo PHP extension
ext-filter                         8.2.7    The filter PHP extension
ext-ftp                            8.2.7    The ftp PHP extension
ext-gd                             8.2.7    The gd PHP extension
ext-gettext                        8.2.7    The gettext PHP extension
ext-gmp                            8.2.7    The gmp PHP extension
ext-hash                           8.2.7    The hash PHP extension
ext-iconv                          8.2.7    The iconv PHP extension
ext-intl                           8.2.7    The intl PHP extension
ext-json                           8.2.7    The json PHP extension
ext-ldap                           8.2.7    The ldap PHP extension
ext-libxml                         8.2.7    The libxml PHP extension
ext-mbstring                       8.2.7    The mbstring PHP extension
ext-mysqli                         8.2.7    The mysqli PHP extension
ext-mysqlnd                        0        The mysqlnd PHP extension (actual version: mysqlnd 8.2.7)
ext-odbc                           8.2.7    The odbc PHP extension
ext-openssl                        8.2.7    The openssl PHP extension
ext-pcntl                          8.2.7    The pcntl PHP extension
ext-pcre                           8.2.7    The pcre PHP extension
ext-pdo                            8.2.7    The PDO PHP extension
ext-pdo_dblib                      8.2.7    The pdo_dblib PHP extension
ext-pdo_mysql                      8.2.7    The pdo_mysql PHP extension
ext-pdo_odbc                       8.2.7    The PDO_ODBC PHP extension
ext-pdo_pgsql                      8.2.7    The pdo_pgsql PHP extension
ext-pdo_sqlite                     8.2.7    The pdo_sqlite PHP extension
ext-pgsql                          8.2.7    The pgsql PHP extension
ext-phar                           8.2.7    The Phar PHP extension
ext-posix                          8.2.7    The posix PHP extension
ext-pspell                         8.2.7    The pspell PHP extension
ext-random                         8.2.7    The random PHP extension
ext-readline                       8.2.7    The readline PHP extension
ext-reflection                     8.2.7    The Reflection PHP extension
ext-session                        8.2.7    The session PHP extension
ext-shmop                          8.2.7    The shmop PHP extension
ext-simplexml                      8.2.7    The SimpleXML PHP extension
ext-soap                           8.2.7    The soap PHP extension
ext-sockets                        8.2.7    The sockets PHP extension
ext-sodium                         8.2.7    The sodium PHP extension
ext-spl                            8.2.7    The SPL PHP extension
ext-sqlite3                        8.2.7    The sqlite3 PHP extension
ext-sysvmsg                        8.2.7    The sysvmsg PHP extension
ext-sysvsem                        8.2.7    The sysvsem PHP extension
ext-sysvshm                        8.2.7    The sysvshm PHP extension
ext-tidy                           8.2.7    The tidy PHP extension
ext-tokenizer                      8.2.7    The tokenizer PHP extension
ext-xml                            8.2.7    The xml PHP extension
ext-xmlreader                      8.2.7    The xmlreader PHP extension
ext-xmlwriter                      8.2.7    The xmlwriter PHP extension
ext-xsl                            8.2.7    The xsl PHP extension
ext-zend-opcache                   8.2.7    The Zend OPcache PHP extension
ext-zip                            1.21.1   The zip PHP extension
ext-zlib                           8.2.7    The zlib PHP extension
lib-bz2                            1.0.8    The bz2 library
lib-curl                           8.1.2    The curl library
lib-curl-(securetransport) openssl 3.1.1    curl (securetransport) openssl version (3.1.1)
lib-curl-libssh2                   1.11.0   curl libssh2 version
lib-curl-zlib                      1.2.11   curl zlib version
lib-date-timelib                   2022.07  date timelib version
lib-date-zoneinfo                  2023.3   zoneinfo ("Olson") database for date
lib-fileinfo-libmagic              540      fileinfo libmagic version
lib-gd                             2.3.3    The gd library
lib-gmp                            6.2.1    The gmp library
lib-icu                            73.2     The ICU unicode and globalization support library
lib-icu-cldr                       43.1     ICU CLDR project version
lib-icu-unicode                    15.0.0   ICU unicode version
lib-icu-zoneinfo                   2023.3   zoneinfo ("Olson") database for icu
lib-ldap-openldap                  2.6.4    OpenLDAP version of ldap
lib-libsodium                      1.0.18   The libsodium library
lib-libxml                         2.9.13   libxml library version
lib-libxslt                        1.1.35   The libxslt library
lib-libxslt-libxml                 2.9.13   libxml version libxslt is compiled against
lib-mbstring-libmbfl               1.3.2    mbstring libmbfl version
lib-mbstring-oniguruma             6.9.8    mbstring oniguruma version
lib-openssl                        3.1.1    OpenSSL 3.1.1 30 May 2023
lib-pcre                           10.42    The pcre library
lib-pcre-unicode                   14.0.0   PCRE Unicode version support
lib-pdo_pgsql-libpq                15.3     libpq for pdo_pgsql
lib-pdo_sqlite-sqlite              3.42.0   The pdo_sqlite-sqlite library
lib-sqlite3-sqlite                 3.42.0   The sqlite3-sqlite library
lib-zip-libzip                     1.10.0   The zip-libzip library
lib-zlib                           1.2.11   The zlib library
php                                8.2.7    The PHP interpreter
php-64bit                          8.2.7    The PHP interpreter, 64bit
php-ipv6                           8.2.7    The PHP interpreter, with IPv6 support
```

### Steps to reproduce the issue.

1. Install xcode beta
2. Install iOS 17 beta
3. Run the following script (replace all the fields):
```<?php
use Kreait\Firebase\Factory;
use Kreait\Firebase\Messaging\ApnsConfig;

require __DIR__ . '/vendor/autoload.php';

$config = [
    'server_key'                  => 'xxx',
    // google auth json contents
    "type"                        => "xxx",
    "project_id"                  => "xxx",
    "private_key_id"              => "xxx",
    "private_key"                 => "xxx",
    "client_email"                => "xxxx",
    "client_id"                   => "xxx",
    "auth_uri"                    => "xxx",
    "token_uri"                   => "xxx",
    "auth_provider_x509_cert_url" => "xxx",
    "client_x509_cert_url"        => "xxx",
];

$factory = (new Factory)->withServiceAccount($config);
$messaging = $factory->createMessaging();

$apnsConfig = ApnsConfig::fromArray([
    'headers' => [
        'apns-priority' => '10',
        'apns-id' => 'xxx',
    ],
    'payload' => [
        'aps' =>
        [
            'alert' => [
                'title' => 'test message with id',
                'body' => 'test message body with id',
            ],
            'sound' => 'default',
            'id'        => '12345',
            'send_date' => date('c'),
        ],
    ],
]);

$deviceToken = 'xxx';
$message = \Kreait\Firebase\Messaging\CloudMessage::withTarget('token', $deviceToken)
->withApnsConfig($apnsConfig)
;
$messaging->send($message);```

### Error message/Stack trace

```shell
Spam of 10 12 push messsages
```

### Additional information

As I said, it works fine on iOS 16

**Comment by jeromegamez** [comment]
I don't have an iOS App to test this myself, but from the code you provided, I cannot see how this could cause spam because you're sending just one message to the FCM service (unless there's a loop that I don't see there).

But if it works on iOS 16, I must assume it has something to do with iOS 17, or that the FCM service does something that it shouldn't do.

Unfortunately, I can't control what Firebase does with the messages or how the client devices handle the messages. And since this is an unofficial SDK, I also don't have contacts at Google to forward this issue to.

Long story short: I don't know how I can provide support here 😕

**Comment by matheuscuba** [comment]
Hello, we have the same issue using a different SDK of Firebase. It seems that something is wrong with the iOS 17 Beta

**Comment by jeromegamez** [comment]
Thank you for chiming in, @matheuscuba! I will close the issue then, let's see how things look when iOS 17 is out of beta 🤞🏻

**Comment by matheuscuba** [comment]
I've created a thread on the Apple Developer Forum: 
https://developer.apple.com/forums/thread/735807
